dex/src/state.rs

```diff
@@ -3411,6 +3497,98 @@ impl State {
         Ok(())
     }
 
```

Comment:
I think one thing that could be done here is to move this repeated logic where you mutate the open orders account to a separate function but also I think fine as is

Reply:
fixed this